src/app/records/plugset-helpers.ts

```diff
@@ -15,19 +15,40 @@ export function itemsForCharacterOrProfilePlugSet(
   );
 }
 
-export function filterDimPlugsAvailableToCharacerOrProfile(
+/**
+ * The set of plug item hashes that are unlocked in the given plugset by the given character.
+ */
+export function unlockedItemsForCharacterOrProfilePlugSet(
+  profileResponse: DestinyProfileResponse,
+  plugSetHash: number,
+  characterId: string
+): Set<number> {
+  const unlockedPlugs = new Set<number>();
+  const plugSetItems = itemsForCharacterOrProfilePlugSet(profileResponse, plugSetHash, characterId);
+  // TODO: would be great to precalculate/memoize this by character ID and profileResponse
+  for (const plugSetItem of plugSetItems) {
+    // Only items that can be inserted are unlocked
+    if (plugSetItem.canInsert) {
+      unlockedPlugs.add(plugSetItem.plugItemHash);
+    }
+  }
+  return unlockedPlugs;
+}
+
+/**
+ * Narrow down the passed in plugSet's plugs to only those that are unlocked by the given character.
+ */
+export function filterDimPlugsUnlockedOnCharacterOrProfile(
```

Comment:
I was honestly confused as to what this function was trying to do before, and it seemed like a bug to me that it would look up plugSets in profile data but didn't look at whether they were actually unlocked. I changed it to filter down to unlocked plugs but let me know if that wasn't the intent.

Reply:
Probably me not understanding the link between profile data and unlocked status. It was intended to be a wrapper around the `itemsForCharacterOrProfilePlugSet` function which checked to see if plugs were available on the profile data.

I had assumed that we could have plugs in the definitions, but not in the profile. So this was intended to filter the plugs in the defs down to those in the profile.

I think I was checking `canInsert` after this function was used and if I wasn't that is a bug.